### PR TITLE
Bug/dsg 696/add arraylist to localdate converter

### PR DESCRIPTION
**Pull request by jgnuvalence**
## Types of changes

Patch
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Documentation change (documentation only, will not require redeployment)
- [ ] Refactor or cleanup (functionality unchanged)
Minor
- [ ] New feature (non-breaking change which adds functionality)
Major
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [ ] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have performed a self-review of my own code
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] Any dependent changes have been merged and published in downstream modules

- [ ] I have confirmed all acceptance criteria defined by the story have been met

### CODE REVIEW

**Comment by jgnuvalence** [comment]
It is expected that for now the checks of this PR will fail since version `0.1.2` of `dsgov-auth` does not exist yet, this PR must first be merged:
[](https://github.com/Nuvalence/dsgov-libs/pull/14)

and must also release the new version

File: service/src/main/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupport.java [hunk 1]
```diff
@@ -193,4 +199,16 @@ private static Document convertMapToDocument(Map<String, Object> documentMap) {
                 .questionKey(questionKey)
                 .build();
     }
+
+    private static LocalDate convertArrayListToLocalDate(ArrayList<Integer> value) {
+        final int dateArraySize = 3;
+        if (value.size() < dateArraySize) {
```
Comment (asultan001) [inline service/src/main/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupport.java:205]: should this be !=  instead? do we ever expect more values to be passed in the array? Maybe time related values would be sent?

File: service/src/test/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupportTest.java [hunk 2]
```diff
@@ -35,6 +37,22 @@ void convertStringToLocalTime() {
         assertNull(DataConversionSupport.convert("", LocalTime.class));
     }
 
+    @Test
+    void convertArrayListToLocalDate() {
```
Comment (asultan001) [inline service/src/test/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupportTest.java:41]: Can we add some negative tests here also? what happens when negative numbers are sent or days and months are greater than 31 and 12? what happens when you try to add a day that doesn't exist on a month like Feb 31st? etc.

Comment (jgnuvalence) [reply service/src/test/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupportTest.java:41]: @asultan001  this arrayList is created by jackson when saving a LocalDate as json, this is not an array created by us, so this is not supposed to happen

File: service/src/main/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupport.java (hunk 1 above)
Comment (jgnuvalence) [reply service/src/main/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupport.java]: I just pushed a fix for this @asultan001

File: service/src/test/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupportTest.java (hunk 2 above)
Comment (asultan001) [reply service/src/test/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupportTest.java:41]: Got it. In that case then it probably doesn't require that level of validation. My thought though was that given this is an API, anyone could call the API with any info correct? or is this being converted after it reaches the API?

File: service/src/main/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupport.java [hunk 3]
```diff
@@ -193,4 +200,22 @@ private static Document convertMapToDocument(Map<String, Object> documentMap) {
...
     }
+
+    private static LocalDate convertArrayListToLocalDate(ArrayList<Integer> value) {
+        try {
+            final int dateArraySize = 3;
+            if (value.size() != dateArraySize || value.contains(null)) {
+                throw new DateTimeException("Malformed date");
+            }
+
+            int year = value.get(0);
```
Comment (JayStGelais) [inline service/src/main/java/io/nuvalence/workmanager/service/domain/dynamicschema/DataConversionSupport.java:211]: Why are we useing a list of Integers to submit data data.

Can't the front-end send a string in YYYY-MM-DD format?
